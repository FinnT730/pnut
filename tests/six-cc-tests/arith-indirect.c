void putstring(char * s) {
  while (*s) {
    putchar(*s);
    s = s + 1;
  }
}

int main() {
  int * arr;
  int acc;

  arr = malloc(100);

  int i = 0;
  while(i < 100) {
    arr[i] = i;
    i = i + 1;
  }

  acc = 0;
  while(i < 100) {
    acc += arr[i] * 13;
    i = i + 1;
  }
  //printf("acc: %d\n", acc);
  putstring("acc: ");
  putchar(acc);
  putchar('\n');

  acc = 0;
  int i = 0;
  while (i < 100) {
    acc += (*(arr + i)) * 13;
    i = i + 1;
  }
  //printf("acc: %d\n", acc);
  putstring("acc: ");
  putchar(acc);
  putchar('\n');
  return 0;
}
