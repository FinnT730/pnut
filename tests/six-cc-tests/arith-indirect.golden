acc: 64350
acc: 64350
